impl/src/main/java/io/kroxylicious/testing/kafka/common/Version.java

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.kroxylicious.testing.kafka.common;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ ElementType.FIELD, ElementType.PARAMETER })
+public @interface Version {
+    String[] values();
```

Comment:
Should `@Version` support a range of values, or specify a single one? Compare with the other annotations where `@BrokerCluster` can't express a range of cluster sizes, for example. Instead we test over a range using `@DimensionMethodSource`, and by specifying multiple dimensions we test over the Cartesian product. 

The reason I did it like that is it's a bit more flexible. Specifying multiple values is fine when the single thing is a scalar, like version. But it doesn't work well you want to test over a range of cases where each case has to be specified as a more complex type such as sets of broker configs.

Reply:
My intention was to set `@Version` as a parameter and then in the context provider use those range of values to launch the same tests for the versions indicated in the array. Inside the context provider I did different combinations with the brokersNum, KRaft/Zookeeper mode... as they were covered in the integration tests, but I can try to follow your approach also with the `@TestTemplate`

Reply:
Done